### ISSUE TITLE
input of run_de function when using a Seurat object

### ISSUE THREAD

**Issue by pandaqiuqiu** [neurorestore/Libra#37]
Thanks for your useful tools for scRNA analysis!
I just want to know for sure about input of run_de function when using a Seurat object:

Is input raw counts or "RNA" assay when de_family = "singlecell"? 
Is input only raw counts when de_family = "pseudobulk"? 
What about "mixedmodel"? 

I appreciate it if you answer this question.

**Comment by jordansquair** [comment]
By default it will use "RNA" as the assay. You can always pass the count matrix in directly.

Yes of course pseudobulk and mixedmodel use counts only.

**Comment by pandaqiuqiu** [comment]
Make sense!  Thanks so much.
Another question is, 
If I set label as input of label_col like this: label = factor(label, levels=c("A", "B")), and then run: run_de(expr, meta, label_col = label).   Does the avg_logFC of the result mean A vs B (as reference) or not?

**Comment by jordansquair** [comment]
The first level of the factor would be used as the reference. so levels = c("A", "B") with a logFC > 1 means B > A.

**Comment by pandaqiuqiu** [comment]
> The first level of the factor would be used as the reference. so levels = c("A", "B") with a logFC > 1 means B > A.

Thanks for pointing out my mistake. I really appreciate that.

**Comment by pandaqiuqiu** [comment]
I just encountered something confusing. As you have described, I used the following to calculate B vs A (A as reference), with code and result as below.

seurat$TimePoint <- factor(seurat$TimePoint,levels = c("A","B"))
DE = run_de(seurat, 
            cell_type_col = "Celltype", 
            replicate_col="Sample", 
            label_col = "TimePoint",          
            de_family = "pseudobulk",         
            de_method = "DESeq2",             
            de_type = "LRT",                 
            n_threads = 4)

![image](https://user-images.githubusercontent.com/89887892/223228695-260fb814-9444-4181-a999-0de94ff2df33.png)

But when I used the same data as input and same method(deseq2+LRT) following as tutorial 
(https://github.com/hbctraining/scRNA-seq_online/blob/master/lessons/pseudobulk_DESeq2_scrnaseq.md#scatterplot-of-normalized-expression-of-top-20-most-significant-genes), the logFC result is reverse, the key code and results is like this: 

dds$group_id = relevel(dds$group_id, ref = "A")
dds <- DESeq(dds, test="LRT", reduced=~1)

![image](https://user-images.githubusercontent.com/89887892/223229126-cd76ab61-9a2c-4f6d-884a-d8ff5e6c9eed.png)

Can you take a look to see what's causing the problem?

**Comment by jordansquair** [comment]
I'm not exactly sure from your post what the problem is. You can just plot some of these genes to see if the direction is correct no?

**Comment by pandaqiuqiu** [comment]
Yes.
I used to_pseudobulk() function to get counts at sample-level for plotting A1BG gene. The stat and plot like this:
![image](https://user-images.githubusercontent.com/89887892/223286974-e639b706-9de0-45f7-87f7-81f1dd564b61.png)
![image](https://user-images.githubusercontent.com/89887892/223286989-0401c023-6197-4d1e-bfb2-f8c2301b02d4.png)

 
Also, I used plotCounts(dds) function to get normalized data at sample-level for plotting A1BG gene. The stat and plot like this:
![image](https://user-images.githubusercontent.com/89887892/223286923-adf8411a-9437-4ce9-83ae-05398048c062.png)
![image](https://user-images.githubusercontent.com/89887892/223286939-470c8983-e9b7-4e5f-877a-fa93cfa5a2fa.png)

It seems A1BG is down-regulated in B compared A, so log2FC should be negative in result. What do you think about it?

**Comment by jordansquair** [comment]
Yes sorry - typo above -- the second factor level is used as reference.

**Comment by pandaqiuqiu** [comment]
> Yes sorry - typo above -- the second factor level is used as reference.

Thanks. Good to have this figured out.
